src/main/java/core/basesyntax/strategy/handler/BalanceOperationHandler.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.strategy.handler;
+
+import core.basesyntax.dao.FruitDao;
+import core.basesyntax.model.FruitTransaction;
+
+public class BalanceOperationHandler implements OperationHandler {
```

Comment:
The class 'BalanceOperationHandler' should not directly manipulate the storage map from 'fruitDao'. Instead, it should use methods provided by the 'FruitDao' interface to ensure encapsulation is maintained.